xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/XMLUtils.java

```diff
@@ -349,24 +439,44 @@ public static String escapeElementContent(Object content)
      * Unescape encoded special XML characters. Only &gt;, &lt; &amp;, ", ' and { are unescaped, since they are the only
      * ones that affect the resulting markup.
      *
-     * @param content the text to decode, may be {@code null}
+     * @param content the text to decode, may be {@code null}. The content is converted to {@code String} using
+     * {@link String#valueOf(Object)} before escaping.
      * @return unescaped content, {@code null} if {@code null} input
+     * @deprecated since 12.8RC1, use {@link #unescape(String)} instead
      */
+    @Deprecated
     public static String unescape(Object content)
     {
         if (content == null) {
             return null;
         }
-        String str = String.valueOf(content);
 
-        str = APOS_PATTERN.matcher(str).replaceAll("'");
-        str = QUOT_PATTERN.matcher(str).replaceAll("\"");
-        str = LT_PATTERN.matcher(str).replaceAll("<");
-        str = GT_PATTERN.matcher(str).replaceAll(">");
-        str = AMP_PATTERN.matcher(str).replaceAll("&");
-        str = LCURL_PATTERN.matcher(str).replaceAll("{");
+        return unescape(String.valueOf(content));
+    }
+
+    /**
+     * Unescape encoded special XML characters. Only &gt;, &lt; &amp;, ", ' and { are unescaped, since they are the only
```

Comment:
There is no reason to provide a custom unecaper, anyone with the need to unecape XML should use one of the many existing XML unescaper.

Reply:
Unescaper deprecated and `org.apache.commons.text.StringEscapeUtils#unescapeXml(String)` suggested insted.